src/Doctrine/ReadModel/CacheDocumentRepository.php

```diff
@@ -35,17 +38,24 @@ public function fetch(string $id, bool $includeMetadata = false): JsonDocument
         return new JsonDocument($id, $value);
     }
 
-    public function save(JsonDocument $document): void
+    public function save(JsonDocument $document, int $attempts = 3): void
```

Comment:
Maybe also make this config? You can still default it to 3 like you defaulted the milliseconds to 0. That way we can easily experiment with it. And there is no reason to have a different number of attempts depending on the caller of this function I think.

Reply:
The allowed retries are no injected in the constructor.